services/provider/server/server.go

```diff
@@ -1266,3 +1281,17 @@ func (s *OCSProviderServer) isSystemInMaintenanceMode(ctx context.Context) (bool
 	}
 	return kerrors.IsNotFound(err), nil
 }
+
+func (s *OCSProviderServer) isConsumerMirrorEnabled(ctx context.Context, consumer *ocsv1alpha1.StorageConsumer) (bool, error) {
+	clientMappingConfig := &corev1.ConfigMap{}
+	clientMappingConfig.Name = util.StorageClientMappingConfigName
+	clientMappingConfig.Name = s.namespace
+	err := s.client.Get(ctx, client.ObjectKeyFromObject(clientMappingConfig), clientMappingConfig)
+	if client.IgnoreNotFound(err) != nil {
+		return false, err
+	} else if kerrors.IsNotFound(err) {
+		return false, nil
+	}
+
+	return clientMappingConfig.Data != nil && clientMappingConfig.Data[consumer.Status.Client.ID] != "", nil
```

Comment:
No need for the first check. In go a nil map can still be queried 
```suggestion
	return clientMappingConfig.Data[consumer.Status.Client.ID] != "", nil
```